torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -1031,6 +1030,278 @@ std::pair<TensorDomain*, TensorDomain*> TensorDomain::rFactor(
       TransformRFactor::runReplay2(this, axes)};
 }
 
+namespace {
+
+/** \brief Container class DisjointSet models equivalence relationships
+ *  \details each instance of this class keeps a set of equivalent classes
+ *           DisjointSet::join(a,b) makes the full class of a and b equivalent
+ *           DisjointSet::areEqual(a,b) checks if a and b belong same class
+ *
+ */
+template <typename T>
+class DisjointSet {
+ public:
+  DisjointSet() = default;
+
+  /** \brief Joins the equivalent class that a and b belong to
+   *         areEqual(a',b') will be true for each a'=a and b'=b
+   */
+  void join(T a, T b) {
```

Comment:
can this be `join(const T a, const Tb)`?

Reply:
Thanks for the suggestions!

Reply:
Here, `Val*` is used `T`, so the signature is `void join(Val* a, Val* b)`. Or, perhaps, `DisjointSet` is used as `DisjointSet<const Val*>`. It's just not intuitive for me that joining sets corresponding to some values imply those values could be modified.